staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -79,7 +79,7 @@ func Parse(val string) IntOrString {
 	if err != nil {
 		return FromString(val)
 	}
-	return FromInt(i)
+	return FromInt32(int32(i))
```

Comment:
this probably bears explanation, why we're accepting the overflow instead of preserving it as a string

I assume the only reason we would do this is to preserve existing behavior, but otherwise, the string seems preferable

Reply:
Yes I think what I did was having a behavior change, I revised to a newer version to check the overflow. PTAL, thanks!

Reply:
you can drop the `if i > math.MaxInt32 || i < math.MinInt32` check and just switch the `strconv.Atoi(val)` call to `strconv.ParseInt(val, 10, 32)`, which will handle the overflow errors and make sure the parsed result fits in 32 bits, regardless of the platform

Reply:
also update the documentation to `Parse the given string and try to convert it to an int32...`

Reply:
Fixed, thanks